### ISSUE TITLE
AvDateRange breaks on input field changes

### ISSUE THREAD

**Issue by KaseyPowers** [Availity/availity-react#40]
In AvDateRange, when attempting to update an input field manually it becomes `[Object object]`
![screen shot 2019-01-31 at 3 22 24 pm](https://user-images.githubusercontent.com/4523739/52082923-188b7880-256c-11e9-927d-2d93d06e9b34.png)

It looks like whats happening is that AvDate's onChange handler is called after the event expires, meaning the event values are now null
https://github.com/Availity/availity-react/blob/master/packages/reactstrap-validation-date/AvDate.js#L125

I'm not sure why that call happens in setState like this, it seems like if it was moved into the main function, the event destructuring in AvDateRange should work as expected. Otherwise I suspect we should either use `event.persist()` or have AvDateRange use the second value argument instead.

**Comment by jordan-a-young** [comment]
I encountered the same issue and overrode the `onStartFieldChange` and `onEndFieldChange` to fix it. Below are the resulting methods: 

```
  onStartFieldChange = (event, target) => {
    const { endValue } = this.state;
    const { onChange } = this.props;

    const startValue = event && target ? target : '';
    this.setState({ startValue, open: false }, () => {
      if (onChange) {
        onChange(event, {
          start: startValue,
          end: endValue,
        });
      }
    });
  };

  onEndFieldChange = (event, target) => {
    const { startValue } = this.state;
    const { onChange } = this.props;

    const endValue = event && target ? target : '';
    this.setState({ endValue, open: false }, () => {
      if (onChange) {
        onChange(event, {
          start: startValue,
          end: endValue,
        });
      }
    });
  };
```

**Comment by TheSharpieOne** [comment]
I forget why I put the onChange in the setState callback... but I believe there was a reason... so I would suggest maybe using `event.persist()`

**Comment by robmcguinness** [comment]
fixed in #43
